docs/src/tutorials/getting_started/transitioning_from_matlab.jl

```diff
@@ -0,0 +1,380 @@
+# [`primal_status`](@ref), and [`raw_status`](@ref), See [Solutions](@ref jump_solutions)
+# for more details on how to query and interpret solution statuses.
+
+# ## Extracting variables
+
+# Like YALMIP, but unlike CVX, with JuMP you need to explicitly ask for the value
+# of your variables after optimization is done, with the function call `value(x)`
+# to obtain the value of variable `x`.
+
+value.(m[1][1, 1])
+
+# A subtlety is that, unlike YALMIP, the function `value` is only defined for
+# scalars. For vectors and matrices you need to use Julia broadcasting:
+# `value.(v)`.
+
+value.(m[1])
+
+# There is also a specialized function for extracting the value of the objective,
+# `objective_value(model)`, which is useful if your objective doesn't have a
+# convenient expression.
+
+objective_value(model)
+
+# ## Dual variables
+
+# Like YALMIP and CVX, JuMP allows you to recover the dual variables. In order
+# to do that, the simplest method is to name the constraint you're interested in,
+# for example, `@constraint(model, bob, sum(v) == 1)` and then, after the
+# optimzation is done, call `dual(bob)`. See [Duality](@ref) for more details.
+
+# ## Reformulating problems
+
+# Perhaps the biggest difference between JuMP and YALMIP and CVX is how far the
+# package is willing to go in reformulating the problems you give to it.
+
+# CVX is happy to reformulate anything it can, even using approximations if your
+# solver cannot handle the problem.
+
+# YALMIP will only do exact reformulations, but is still fairly adventurous,
+# for example, being willing to reformulate a nonlinear objective in terms of
+# conic constraints.
+
+# JuMP does no such thing: it only reformulates objectives into objectives, and
+# constraints into constraints, and is fairly conservative at that. As a result,
+# you might need to do some reformulations manually, for which a good guide is
+# the [Tips and tricks](@ref conic_tips_and_tricks) tutorial.
+
+# ## Vectorization
+
+# In MATLAB, it is absolutely essential to "vectorize" your code to obtain
+# acceptable performance. This is because MATLAB is a slow interpreted
+# language, which sends your commands to fast libraries. When you "vectorize"
+# your code you are minimizing the MATLAB part of the work and sending it to the
+# fast libraries instead.
+
+# There's no such duality with Julia.
+
+# Everything you write and most libraries you use will compile down to LLVM, so
+# "vectorization" has no effect.
+
+# For example, if you are writing a linear program in MATLAB and instead of the
+# usual `constraints = [v >= 0]` you write:
+# ```matlab
+# for i = 1:n
+#    constraints = [constraints, v(i) >= 0];
+# end
+# ```
+# performance will be poor.
+
+# With Julia, on the other hand, there is hardly any difference between
+# ```julia
+# @constraint(model, v >= 0)
+# ```
+#  and
+# ```julia
+# for i in 1:n
+#     @constraint(model, v[i] >= 0)
+# end
+# ```
+
```

Comment:
```suggestion

# ## Arbitrary precision

# JuMP is capable of both specifying the optimization problem and solving it in a
# great variety of types, allowing it to work in single, double, quadruple, or 
# arbitrary precision, for example. In contrast, neither YALMIP nor CVX can do
# this, as they are restricted by MATLAB's archaic type system (with the package
# [mpYALMIP](https://github.com/aeroimperial-optimization/mpYALMIP) one can
# modify YALMIP to support solving in multiple precision, but still the problem
# can only be specified in double precision).

# In order to do that, one needs to create the model with the command `model = GenericModel{T}()`,
# where `T` is the type you want to use, for example `BigFloat`, and then set the
# optimizer with the command `set_optimizer(model, Hypatia.Optimizer{T})`. At the
# time of writing the solvers supporting generic types are [Clarabel](https://github.com/oxfordcontrol/Clarabel.jl),
# [COSMO](https://github.com/oxfordcontrol/COSMO.jl), and [Hypatia](https://github.com/jump-dev/Hypatia.jl).

# See [here](https://jump.dev/JuMP.jl/stable/tutorials/conic/arbitrary_precision/)
# for a complete guide on using JuMP with generic types.

```

Reply:
I'm not sure about this part. We should focus on the core features of the packages that are similar.

There's the arbitrary precision tutorial for experts. But that's not something they need to know when they first transition from MATLAB.

Reply:
Both the similarities and the differences are necessary, and this is a key difference. MATLAB users won't dream that this this is possible, so they won't find the arbitrary precision tutorial. This section is just intended to take them there.

Reply:
I don't want to add this to a a "Transitioning from MATLAB" tutorial. Also, the arbitrary precision is not a panacea. Users should not just use a solver with more precision if they run into numerical difficulties.

Reply:
The point of arbitrary precision is not to solve numerical problems, but to obtain higher precision. In my [latest paper](https://arxiv.org/abs/2311.18707) I had to use an arbitrary precision solver because MOSEK just couldn't give me 8 digits, even when I tweaked its precision, and I really needed 8 digits. With SDPA-GMP I could easily get more than 20 digits. Ironically enough, SDPA-GMP was much more vulnerable than MOSEK to lack of strict feasibility.

Reply:
I understand your point. But it doesn't belong in this tutorial. If there is an argument to be made, it is that we should make the arbitrary precision more widely known for _all_ users, not just those coming from MATLAB.